hilti/toolchain/include/compiler/global-optimizer.h

```diff
@@ -21,12 +21,13 @@ struct GlobalOptimizer {
     using FieldID = ID;
 
     struct Uses {
+        bool hook = false;
         bool declared = false;
         bool defined = false;
         bool referenced = false;
     };
 
-    using Hooks = std::map<std::tuple<ModuleID, StructID, FieldID>, Uses>;
+    using Functions = std::map<std::tuple<ModuleID, StructID, FieldID>, Uses>;
```

Comment:
This is ok, but as food for thought: I'm back to thinking that my earlier suggestion of using fully qualified/normalized IDs instead of tuples might work well to capture both hooks and functions, without then having to special-case whether the struct ID is available or not. (But fine leaving as is.)

Reply:
I am leaving this as is for now.